src/Enzyme.jl

```diff
@@ -151,13 +153,32 @@ while ``\\partial f/\\partial b`` will be *added to* `∂f_∂b` (but not return
     [`Active`](@ref) will automatically convert plain integers to floating
     point values, but cannot do so for integer values in tuples and structs.
 """
-@inline function autodiff(f::F, args...) where F
+@inline function autodiff(f::F, ::Type{A}, args...) where {F, A<:Annotation}
+    args′  = annotate(args...)
```

Comment:
I'm still inclined to have the API have the annotation automatically deduced by default if theres a nice way to do that here.

Reply:
not really, we can't have a default argument before varargs. We could allow for some ambiguity and check if the first vararg is an Annotation type.

Reply:
What if the first argument was the annotation as the specialization, and if it wasn't an annotation we automatically deduces and calls the other version?